src/runner/bootstrapper.js

```diff
@@ -31,15 +46,35 @@ export default class Bootstrapper {
         return flatten(browserInfo);
     }
 
-    _createConnectionFromBrowserInfo (browserInfo) {
-        if (browserInfo instanceof BrowserConnection)
-            return browserInfo;
+    _createAutomatedConnections (browserInfo) {
+        if (!browserInfo)
+            return [];
+
+        var browserConnections = [];
+
+        for (var i = 0; i < this.concurrency; i++) {
+            var nextConnections = browserInfo.map(browser => new BrowserConnection(this.browserConnectionGateway, browser));
 
-        return new BrowserConnection(this.browserConnectionGateway, browserInfo);
+            browserConnections = zipWith(browserConnections, nextConnections, (prev, next) => {
+                if (!Array.isArray(prev))
+                    return [next];
+
+                return prev.concat(next);
+            });
+        }
+
+        return browserConnections;
```

Comment:
As I understand this function transforms args in the following way:
```
// concurrency === 2
['chrome', 'ff'] -> [[chromeBC, chromeBC], [ffBC, ffBC]]
```
Am I right? If yes, why it's too complicated logic with `zipWith` here. Why not just something like
```js

if (!browserInfo)
    return [];

return browserInfo.map(browser => {
    var connections = [];

    for(let i = 0; i < this.concurrencty.length; i++)
        connections.push(new BrowserConnection(this.browserConnectionGateway, browser));

    return connections;
});
```